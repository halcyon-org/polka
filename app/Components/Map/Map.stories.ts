import type { Meta, StoryObj } from "@storybook/react";

import { Map } from "./Map";

type T = typeof Map;

const meta: Meta = {
  title: "Map",
  component: Map,
  parameters: {
    layout: "fullscreen",
  },
  tags: ["autodocs"],
} satisfies Meta<T>;

type Story = StoryObj<T>;

export const Default: Story = {};

export default meta;
