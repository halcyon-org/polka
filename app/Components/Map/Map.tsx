import { type ReactNode } from "react";
import { GoogleMap, LoadScript } from "@react-google-maps/api";

const containerStyle = {
  width: "108rem",
  height: "50.7rem",
};

const center = {
  lat: 34.8503, // 東京の緯度
  lng: 136.582, // 東京の経度
};

type MapComponentProps = {
  apiKey: string;
};

const MapComponent = ({ apiKey }: MapComponentProps): ReactNode => {
  return (
    <LoadScript googleMapsApiKey={apiKey}>
      <GoogleMap
        mapContainerStyle={containerStyle}
        center={center}
        zoom={15}
      ></GoogleMap>
    </LoadScript>
  );
};

export const Map = ({ apiKey }: MapComponentProps): ReactNode => {
  if (!apiKey) {
    return <p>Google Maps APIキーが設定されていません。</p>;
  }

  return (
    <div>
      <MapComponent apiKey={apiKey} />
    </div>
  );
};

export default Map;
