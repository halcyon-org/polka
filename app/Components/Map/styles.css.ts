export const center = {
  lat: 34.8503,
  lng: 136.582,
};
