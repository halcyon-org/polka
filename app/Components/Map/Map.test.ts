// @vitest-environment jsdom

import { expect, test } from "vitest";

import { composeStories } from "@storybook/react";

import * as stories from "./Map.stories";

const { Default } = composeStories(stories);
test("Map snapshot", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
