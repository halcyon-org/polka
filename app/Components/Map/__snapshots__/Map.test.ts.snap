// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Map snapshot 1`] = `
<div>
  <div />
  <div>
    Loading...
  </div>
</div>
`;
